### PR TITLE
Remove collection watchers

### PR DESCRIPTION
**Pull request by n1k0**
This patch is the final step in the refactoring process started with #156, #157, #158 and #159.

### CODE REVIEW

**Comment by n1k0** [comment]
r=? @leplatrem @Natim @magopian @glasserc

File: test/sagas/collection_test.js [hunk 1]
```diff
@@ -101,14 +99,27 @@ describe("collection sagas", () => {
...
   describe("createRecord()", () => {
-    describe("Success", () => {
-      let collection, createRecord;
+    let collection;
+
+    before(() => {
+      collection = {createRecord() {}};
+      const bucket = {collection() {return collection;}};
+      setClient({bucket() {return bucket;}});
+    });
```
Comment (glasserc) [inline test/sagas/collection_test.js]: Why is this a `before` and not a `beforeEach`? When you're testing generators, I totally understand why you only want to initialize things once, but I thought it was somehow better practice to use `beforeEach` if you didn't need to initialize things once.

**Comment by glasserc** [comment]
I'm a little nervous about how tightly the tests are tied to the implementation that satisfies them. It feels a little bit like we're just writing the same code twice, once as an implementation and once as a test. But I do see the value, especially in the case of things like wanting to simulate failure.

Anyhow, r+wq :)

File: test/sagas/collection_test.js (hunk 1 above)
Comment (n1k0) [inline test/sagas/collection_test.js]: It's just as we don't stub anything here, we don't have to reset/revert anything too, so `before` is just enough.

**Comment by n1k0** [comment]
> I'm a little nervous about how tightly the tests are tied to the implementation that satisfies them.

Is this a general remark with the saga tests or with this patch in particular?

I think it's possible to have some generic test helper checking that a given yielded value is part of a sequence, so we could have much less verbose tests, but we'd loose sequential checks, which I think is rather important when testing sequential async flows.

How would you test this better?

**Comment by glasserc** [comment]
It's a general remark with the saga tests. In my opinion, a good test is one that tests behavior against known properties of a correct solution. Instead, these tests just test that the implementation behaves exactly as the implementation does. It feels like there's minimal value in that.

I don't really have any better ideas. At this point, I'd be trying to write integration tests at a coarser level -- for instance, sending a `RECORD_DELETE_REQUEST` and verifying that a record was deleted. Or we could drop this kind of testing and rely on the browser tests you were working on. That would at least examine the behavior of the implementation. But that's outside the scope of this PR.

Another thing that's outside the scope of this PR -- why are you yielding `call`s to `uuid` ? It's not asynchronous, is it? I thought it might be because you wanted to force it to return different results, but then why not just stub/mock the `uuid` function?

**Comment by n1k0** [comment]
> for instance, sending a RECORD_DELETE_REQUEST and verifying that a record was deleted

You mean involving an actual Kinto server running? That would even be functional testing and would get us far beyond unit testing. These unit tests are verbose but they ensure the i/os with the generator state itself are behaving as expected. I don't see anything wrong with this, it's useful to me at least; but I agree I'm finding the tests very verbose. I'll keep thinking about this, but I'm gonna land this patch now.

> why are you yielding calls to uuid ?

Because I'm trying to use `call` for every calls, for consistency and future-proofiness. Also it's easier to test as you don't need to stub it as you mentioned.

**Pull request merged**